specific/cmo_sale/models/sale.py

```diff
@@ -65,7 +80,40 @@ class SaleOrder(models.Model):
         'sale.order',
         string='Ref.Quotation',
         states={'done': [('readonly', True)]},
+        domain=[
+            '&', ('order_type', 'like', 'quotation'),
+            ('state', 'not like', 'cancel'),
+        ],
     )
+    approval_id = fields.Many2one(
+        'res.users',
+        string='Approval',
+        states={'done': [('readonly', True)]},
+    )
+    margin_percentage = fields.Float(
+        string='Margin Percentage (%)',
+        readonly=True,
+        compute='_compute_margin_percentage',
+    )
+
+    @api.model
+    def create(self, vals):
+        ctx = self._context.copy()
+        current_date = datetime.date.today()
```

Comment:
To get date, please better to use
`fields.Date.context_today(self)`
It make sure that the time zone is taken in consideration.